Assets/Scripts/Auction/BiddingPlatform.cs

```diff
@@ -0,0 +1,123 @@
+using System.Collections;
+using System.Collections.Generic;
+using TMPro;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+[RequireComponent(typeof(Rigidbody))]
+[RequireComponent(typeof(Timer))]
+public class BiddingPlatform : MonoBehaviour
+{
+    [SerializeField]
+    private Item item;
+
+    [SerializeField]
+    public int chips = 0;
+
+    [SerializeField]
+    private PlayerIdentity leadingBidder;
+
+    [SerializeField]
+    private TMP_Text itemNameText;
+
+    [SerializeField]
+    private TMP_Text itemDescriptionText;
+
+    [SerializeField]
+    private TMP_Text itemCostText;
+
+    [SerializeField]
+    private TMP_Text timerText;
+
+    [SerializeField]
+    private float baseWaitTime;
+
+    [SerializeField]
+    private float bumpTime = 5.0f;
+
+    public BiddingRound activeBiddingRound;
+
+    [SerializeField]
+    private Timer auctionTimer;
+
+    private void OnTriggerEnter(Collider other)
+    {
+        if (other.TryGetComponent(out PlayerManager playerManager)){
+            playerManager.selectedBiddingPlatform = this;
+        }
+
+    }
+
+    private void OnTriggerExit(Collider other)
+    {
+        if (other.TryGetComponent(out PlayerManager playerManager))
+        {
+            if (playerManager.selectedBiddingPlatform == this)
+            playerManager.selectedBiddingPlatform = null;
+        }
+    }
+
+    private void Start()
+    {
+        auctionTimer = GetComponent<Timer>();
+        auctionTimer.OnTimerUpdate += UpdateTimer;
+        auctionTimer.OnTimerRunCompleted += EndAuction;
+    }
+
+    private void OnDestroy()
+    {
+        auctionTimer.OnTimerUpdate -= UpdateTimer;
+        auctionTimer.OnTimerRunCompleted -= EndAuction;
+    }
+
+    public bool TryPlaceBid(PlayerIdentity playerIdentity)
+    {
+        if(activeBiddingRound == null)
+        {
+            Debug.Log("No active biddingRound on biddingPlatform!");
+            return false;
+        }
+
+        if (playerIdentity.chips > chips)
+        {
+            // Refund
+            if (leadingBidder)
+            {
+                leadingBidder.UpdateChip(chips);
+            }
+            //activeBiddingRound.chips[0] = chips++;
+            chips++;
+            playerIdentity.UpdateChip(-chips); 
+            itemCostText.text = chips.ToString();
+            leadingBidder = playerIdentity;
```

Comment:
I suppose we don't have a border in the color of the leading bidder around the bidding platform now. I think this would be an important usability improvement, perhaps we could add that sort of thing next? 🙏

Reply:
Yup, that and visible gunstats is a big priority I think!